### PR TITLE
feat: trusted signals

### PR DESCRIPTION
**Pull request by iamnewton**
**Description**

Add in `trustedBiddingSignals` part of the proposal

**Additional context (optional)**

https://github.com/WICG/turtledove/blob/main/FLEDGE.md#31-fetching-real-time-data-from-a-trusted-server

**Checklist**
_Have you considered all of the following?_

Please confirm by checking off each item to ensure you've considered it:

- [x] Did you create or update tests? 
- [x] Did you create or update any documentation?

---

Fixes:

### CODE REVIEW

File: docs/interest-groups.html [hunk 1]
```diff
@@ -43,14 +43,18 @@ <h3>Currently Joined</h3>
 		<script type="module">
 			import * as idb from 'https://cdn.jsdelivr.net/npm/idb-keyval@5/dist/esm/index.js';
 			import Fledge from './scripts/api/esm/index.js';
-			const fledge = new Fledge('https://magniteengineering.github.io/fledge.polyfill/iframe.html', true);
+			const fledge = new Fledge('http://localhost:8000/iframe.html', true);
```
Comment (brodrigu) [inline docs/interest-groups.html:46]: you need to move this back right?

File: docs/interest-groups.html [hunk 2]
```diff
@@ -43,14 +43,18 @@ <h3>Currently Joined</h3>
...
 			const customStore = idb.createStore('fledge.v1', 'interest-groups');
 
 			document.getElementById("join-ig").onclick = async function joinGroup() {
 				await fledge.joinAdInterestGroup({
 					owner: document.getElementById('owner').value,
 					name: document.getElementById('name').value,
-					biddingLogicUrl: document.getElementById('url').value
+					biddingLogicUrl: document.getElementById('url').value,
+					trustedBiddingSignalsUrl: 'http://localhost:8000/get-values.json',
```
Comment (brodrigu) [inline docs/interest-groups.html:54]: should this be the `magniteengineering.github.io`?

File: src/frame/utils.js [hunk 3]
```diff
@@ -169,14 +176,18 @@ const getTrustedSignals = async (url, keys, debug) => {
 
 	const isJSON = response => /\bapplication\/json\b/.test(response?.headers?.get('content-type'));
 
+	debug && echo.log(echo.asProcess(`fetching keys from trusted signals url: ${url}`));
 	const response = await fetch(`${url}?${hostname}&keys=${keys.join(',')}`)
 		.then(response => {
```
Comment (brodrigu) [inline src/frame/utils.js:181]: seems strange to mix `promise.then` with `await` how about
```
try {
   const response = await fetch(`${url}?${hostname}&keys=${keys.join(',')}`)
} catch {
   // error handling
}
```

File: docs/interest-groups.html (hunk 2 above)
Comment (iamnewton) [reply docs/interest-groups.html:54]: It will be, but I'm still testing.  Accidentally checked it in.

File: src/frame/utils.js (hunk 3 above)
Comment (iamnewton) [reply src/frame/utils.js:181]: Good catch.

File: src/frame/auction.js [hunk 4]
```diff
@@ -44,7 +44,10 @@ export default async function runAdAuction (conf, debug) {
 	}
 
 	debug && echo.log(echo.asProcess('getting all scores, filtering and sorting'));
-	const [ winner ] = await getScores(filteredBids, conf, debug);
+	const winners = await getScores(filteredBids, conf, debug);
+	const [ winner ] = winners
+		.filter(({ score }) => score > 0)
+		.sort((a, b) => (a.score > b.score) ? 1 : -1);
```
Comment (brodrigu) [inline src/frame/auction.js:50]: A possible future optimization: is filter + sort more expensive than sorting and just checking the highest value for >0?

**Review by brodrigu — APPROVED** [review]
☕

**Pull request merged**